# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1318626409 14400
#      Fri Oct 14 17:06:49 2011 -0400
# Node ID c7fac37a2afc1c54621037852a8e13f85b23c1e3
# Parent  46e68badedb8636a62869c8535b3c745f8d024c8
class: correctly handle parents when structure array has more than one element

* ov-class.cc (octave_class::octave_class): Assign parent classes to
all elements of structure array.

diff --git a/src/ov-class.cc b/src/ov-class.cc
--- a/src/ov-class.cc
+++ b/src/ov-class.cc
@@ -83,17 +83,25 @@ octave_class::octave_class (const octave
           std::string cnm = parent.class_name ();
 
           if (find_parent_class (cnm))
             error ("duplicate class in parent tree");
           else
             {
               parent_list.push_back (cnm);
 
-              map.assign (cnm, parent);
+              if (map.numel () > 1)
+                {
+                  // If MAP has more than one element, put the parent
+                  // class object in each element.
+
+                  map.assign (cnm, Cell (map.dims (), parent));
+                }
+              else
+                map.assign (cnm, parent);
             }
         }
     }
 
   if (! error_state)
     symbol_table::add_to_parent_map (id, parent_list);
 }
 
